.github/workflows/lint.yml

```diff
@@ -0,0 +1,36 @@
+name: Lint
+
+on: push
+
+jobs:
+  run-linters:
+    name: Run linters
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Git repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # all history needed for commitlint to work
```

Comment:
Does this equal to `git fetch origin main`? Or does it fetch all branches?

I wonder if there's a way of fetching only from HEAD to `main`. That'd be the most efficient.

Reply:
I'm currently looking into [already written action](https://github.com/marketplace/actions/commit-linter) that states:

>  It's necessary that you specify the fetch-depth argument to actions/checkout@v2 step. By default they fetch only latest commit of the branch, but we need more commits since we validate a range of commit messages.

Reply:
https://github.com/marketplace/actions/checkout 😢 
> Set fetch-depth: 0 to fetch all history for all branches and tags

It might be more efficient with the default `fetch-depth` and a subsequent:
```
- run: git fetch origin main
```

And even more if we can fetch only commits from the current HEAD down to the PR target (usually `main`'s last commit).
We can optimise that later tho, needs some Git magic.

Reply:
This ☝️ seem to work also, thanks.